### DISCUSSION TITLE
Compiling with USE_PING in Arduino Ide 1.8.13 - fail with error...

### DISCUSSION THREAD

**Discussion by rapi3** [arendst/Tasmota#11586] [Q&A] [answered]
Hello,
This is my first try to compile Tasmota with Arduino Ide 1.8.13 on Linux.
If I try to use:
#define USE_PING
 it fail:
`exit status 1
previous declaration of 'void t_ping_register_pcb()' with 'C++' linkage`

I tried Master and Development and both it produce the same error.
If I don't use ping it compile without any error.

Can somebody have any idea what it is wrong here ?
thank you.

![Screenshot_2021-04-04_23-52-51](https://user-images.githubusercontent.com/18531150/113522467-e2f5d600-95a0-11eb-878a-7ce8e35197f3.png)

![Screenshot_2021-04-04_23-54-48](https://user-images.githubusercontent.com/18531150/113522502-2fd9ac80-95a1-11eb-86bd-957637bdfc9a.png)

**Comment by sfromis** [comment, marked as answer]
Building with the Arduino IDE is no longer supported. See
https://tasmota.github.io/docs/Compile-your-build/
